include/modem_info.h

```diff
@@ -30,26 +30,91 @@ typedef void (*rsrp_cb_t)(char rsrp_value);
 /**@brief LTE link information data. */
 enum modem_info {
 	MODEM_INFO_RSRP,	/**< Signal strength. */
-	MODEM_INFO_BAND,	/**< Current LTE band. */
-	MODEM_INFO_MODE,	/**< Current mode. */
+	MODEM_INFO_CUR_BAND,	/**< Current LTE band. */
+	MODEM_INFO_SUP_BAND,	/**< Supported LTE bands. */
+	MODEM_INFO_AREA_CODE,   /**< Tracking area code. */
+	MODEM_INFO_UE_MODE,	/**< Current mode. */
 	MODEM_INFO_OPERATOR,	/**< Current operator name. */
+	MODEM_INFO_MCC,		/**< Mobile country code. */
+	MODEM_INFO_MNC,		/**< Mobile network code. */
 	MODEM_INFO_CELLID,	/**< Cell ID of the device. */
 	MODEM_INFO_IP_ADDRESS,  /**< IP address of the device. */
 	MODEM_INFO_UICC,	/**< UICC state. */
 	MODEM_INFO_BATTERY,	/**< Battery voltage. */
 	MODEM_INFO_TEMP,	/**< Temperature level. */
 	MODEM_INFO_FW_VERSION,  /**< Modem firmware version. */
-	MODEM_INFO_ICCID,	/**< SIM ICCID */
+	MODEM_INFO_ICCID,	/**< SIM ICCID. */
+	MODEM_INFO_LTE_MODE,	/**< LTE-M support mode. */
+	MODEM_INFO_NBIOT_MODE,	/**< NB-IoT support mode. */
+	MODEM_INFO_GPS_MODE,	/**< GPS support mode. */
 	MODEM_INFO_COUNT,	/**< Number of legal elements in the enum. */
 };
 
-/** @brief Initialize the link information driver.
+/**@brief LTE parameter data. **/
+struct lte_param {
+	u16_t value; /**< The retrieved value. */
+	char string[MODEM_INFO_MAX_RESPONSE_SIZE]; /**< The retrieved value in string format. */
+	char *data_name; /**< The name of the information type. */
+	enum modem_info type; /**< The information type. */
+};
+
+/**@brief Network parameters. **/
+struct network_param {
+	struct lte_param cur_band; /**< Current LTE band. */
+	struct lte_param sup_band; /**< Supported LTE bands. */
+	struct lte_param area_code; /**< Tracking area code. */
+	struct lte_param operator;
```

Comment:
If I add a comment here, the documentation build fails. It might be that "operator" is a reserved word? It builds okay if I change the name ...
@carlescufi @mbolivar Any idea?

Reply:
>  It might be that "operator" is a reserved word?

Indeed, since this is a C++ reserved word, I think this structure member should be renamed. Some customer somewhere will no doubt want to include this header from C++ :).

Reply:
@hmhalvorsen Will you do that?

Reply:
Yes, I will take a look.